modules/flann/include/opencv2/flann/flann_base.hpp

```diff
@@ -79,7 +79,18 @@ NNIndex<Distance>* load_saved_index(const Matrix<typename Distance::ElementType>
     if (fin == NULL) {
         return NULL;
     }
-    IndexHeader header = load_header(fin);
+
+    IndexHeader header;
+    try
+    {
+        header = load_header(fin);
+    }
+    catch (const FLANNException &)
```

Comment:
What is about other exceptions? They still cause leaks.

Perhaps we need RAII-based scope guard for `fin` with cleanup/destructor with `fclose()` call  (new code should use `std::ifstream`)